### PR TITLE
[ixwebsocket] Bump ixwebsocket's zlib requirement to [>=1.2.11 <2] range

### PR DESCRIPTION
**Pull request by uilianries**
Specify library name and version:  **ixwebsocket/any**

- Replace zlib specific version by ranged version.

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by ericLemanissier — APPROVED** [review]

**Review by RubenRBS — DISMISSED** [review]

**Comment by RubenRBS** [comment]
v2 is failing on windows: https://c3i.jfrog.io/c3i/misc-v2/summary.html?json=https://c3i.jfrog.io/c3i/misc-v2/logs/pr/20199/1-windows-msvc/ixwebsocket/10.4.0//summary.json - Not able to repro

**Review by czoido — DISMISSED** [review]

File: recipes/ixwebsocket/all/conanfile.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def layout(self):
 
     def requirements(self):
         if self.options.get_safe("with_zlib", True):
-            self.requires("zlib/1.2.13")
+            self.requires("zlib/[>=1.2.11 <2]")
         if self.options.tls == "openssl":
             self.requires("openssl/1.1.1s")
```
Comment (mayeut) [inline recipes/ixwebsocket/all/conanfile.py:61]:
```suggestion
            self.requires("openssl/1.1.1w")
```
or
```suggestion
            self.requires("openssl/[>=1.1 <4]")
```
depending on OpenSSL 3.x support.

Comment (RubenRBS) [reply recipes/ixwebsocket/all/conanfile.py:61]: Good catch, will set it to 1.1.1 as upstream does not seem to actually test v3 by the looks of their issues

**Comment by uilianries** [comment]
@RubenRBS currently this PR is failing because winsock2.h is missing from `IXWebSocketHttpHeaders.cpp`, resulting in a missing definition. A possible fix would be:

```
diff --git a/recipes/ixwebsocket/all/conanfile.py b/recipes/ixwebsocket/all/conanfile.py
index 6478e82f1..c32b726fd 100644
--- a/recipes/ixwebsocket/all/conanfile.py
+++ b/recipes/ixwebsocket/all/conanfile.py
@@ -119,6 +119,8 @@ class IXWebSocketConan(ConanFile):
             replace_in_file(self, cmakelists,
                                   "ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}",
                                   "ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} RUNTIME DESTINATION bin")
+        # INFO: IXWebSocketHttpHeaders.cpp consumes WSAEINVAL but there is no winsock2.h included
+        replace_in_file(self, os.path.join(self.source_folder, "ixwebsocket", "IXWebSocketHttpHeaders.cpp"), '#include "IXSocket.h"', '#include "IXNetSystem.h"\n#include "IXSocket.h"')
 
     def build(self):
         self._patch_sources()
```

I have no permission to push this change, it's rejected by GH.

File: recipes/ixwebsocket/all/conanfile.py [hunk 2]
```diff
@@ -119,6 +119,8 @@ def _patch_sources(self):
             replace_in_file(self, cmakelists,
                                   "ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}",
                                   "ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} RUNTIME DESTINATION bin")
+        # INFO: IXWebSocketHttpHeaders.cpp consumes WSAEINVAL but there is no winsock2.h included
+        replace_in_file(self, os.path.join(self.source_folder, "ixwebsocket", "IXWebSocketHttpHeaders.cpp"), '#include "IXSocket.h"', '#include "IXNetSystem.h"\n#include "IXSocket.h"')
```
Comment (uilianries) [inline recipes/ixwebsocket/all/conanfile.py:123]:
```suggestion
        replace_in_file(self, os.path.join(self.source_folder, "ixwebsocket", "IXWebSocketHttpHeaders.cpp"), '#include "IXSocket.h"', '#include "IXNetSystem.h"\n#include "IXSocket.h"')
        # INFO: IXHttpClient.cpp consumes WSAEINVAL but there is no winsock2.h included
         replace_in_file(self, os.path.join(self.source_folder, "ixwebsocket", "IXHttpClient.h"), '#include "IXSocket.h"', '#include "IXNetSystem.h"\n#include "IXSocket.h"')
```

File: recipes/ixwebsocket/all/conanfile.py [hunk 3]
```diff
@@ -119,6 +119,10 @@ def _patch_sources(self):
...
+        # INFO: IXHttpClient.cpp consumes WSAEINVAL but there is no winsock2.h included
+         replace_in_file(self, os.path.join(self.source_folder, "ixwebsocket", "IXHttpClient.h"), '#include "IXSocket.h"', '#include "IXNetSystem.h"\n#include "IXSocket.h"')
```
Comment (uilianries) [inline recipes/ixwebsocket/all/conanfile.py:125]:
```suggestion
            # INFO: IXWebSocketHttpHeaders.cpp consumes WSAEINVAL but there is no winsock2.h included
            replace_in_file(self, os.path.join(self.source_folder, "ixwebsocket", "IXWebSocketHttpHeaders.cpp"), '#include "IXSocket.h"', '#include "IXNetSystem.h"\n#include "IXSocket.h"')
            # INFO: IXHttpClient.cpp consumes WSAEINVAL but there is no winsock2.h included
            replace_in_file(self, os.path.join(self.source_folder, "ixwebsocket", "IXHttpClient.h"), '#include "IXSocket.h"', '#include "IXNetSystem.h"\n#include "IXSocket.h"')
```
Ops!

Comment (uilianries) [inline recipes/ixwebsocket/all/conanfile.py:125]:
```suggestion
        # INFO: IXWebSocketHttpHeaders.cpp consumes WSAEINVAL but there is no winsock2.h included
        replace_in_file(self, os.path.join(self.source_folder, "ixwebsocket", "IXWebSocketHttpHeaders.cpp"), '#include "IXSocket.h"', '#include "IXNetSystem.h"\n#include "IXSocket.h"')
        # INFO: IXHttpClient.cpp consumes WSAEINVAL but there is no winsock2.h included
        replace_in_file(self, os.path.join(self.source_folder, "ixwebsocket", "IXHttpClient.h"), '#include "IXSocket.h"', '#include "IXNetSystem.h"\n#include "IXSocket.h"')
```
Ops, bad indentation

**Review by danimtb — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Pull request merged**